addons/web/controllers/export.py

```diff
@@ -432,7 +432,9 @@ def get_fields(self, model, domain, prefix='', parent_name='',
     @http.route('/web/export/namelist', type='json', auth='user', readonly=True)
     def namelist(self, model, export_id):
         export = request.env['ir.exports'].browse([export_id])
-        return self.fields_info(model, export.export_fields.mapped('name'))
+        export_fields_list = export.export_fields.mapped('name')
+        res = self.fields_info(model, export_fields_list)
+        return sorted(res, key=lambda field: export_fields_list.index(field['id']))
```

Comment:
IMO, it is preferable to change `fields_info` instead. Because it returns a list, it implicitly implies that the order is preserved.

Suggestion to add at the end of `fields_info` (I didn't test it, but you get the idea). Also avoid using `index` because it makes the complexity quadratic (not a big deal here, but I always prefer avoid it):
```python
        indexes_dict = {fname: i for i, fname in enumerate(export_fields)}
        return sorted(field_info, key=lambda field_dict: indexes_dict[field_dict['id']])
```

Don't hesitate if my comment is unclear.